src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/Data/CollectionViewGroupInternal.cs

```diff
@@ -545,27 +547,20 @@ internal void AddSubgroupToMap(object nameKey, CollectionViewGroupInternal subgr
         private void RemoveSubgroupFromMap(CollectionViewGroupInternal subgroup)
         {
             Debug.Assert(subgroup != null);
+
             if (_nameToGroupMap == null)
-            {
                 return;
-            }
-            object keyToBeRemoved = null;
 
             // Search for the subgroup in the map.
-            foreach (object key in _nameToGroupMap.Keys)
+            foreach (KeyValuePair<object, WeakReference> item in _nameToGroupMap)
             {
-                WeakReference weakRef = _nameToGroupMap[key] as WeakReference;
-                if (weakRef != null &&
-                    weakRef.Target == subgroup)
+                if (item.Value.Target == subgroup)
                 {
-                    keyToBeRemoved = key;
+                    _nameToGroupMap.Remove(item.Key);
```

Comment:
can you remove while enumerating?

Reply:
Yeah, in Dictionaries it works, magic since .NET Core 3.0 afaik?

Reply:
ah ok, great!